components/camera/replaypcd/replaypcd_utils_test.go

```diff
@@ -324,3 +328,29 @@ func getPointCloudFromArtifact(i int) (pointcloud.PointCloud, error) {
 
 	return pcExpected, nil
 }
+
+type ctrl struct {
+	statusCode    int
+	pcdFileNumber int
+}
+
+func (c *ctrl) mockHandler(w http.ResponseWriter, r *http.Request) {
+	path := fmt.Sprintf(datasetDirectory, c.pcdFileNumber)
+	pcdFile, _ := getCompressedBytesFromArtifact(path)
+
+	w.WriteHeader(c.statusCode)
+	w.Write(pcdFile)
+}
+
+// HTTPMock creates a mock HTTP server.
+func HTTPMock(pattern string, statusCode int) []*httptest.Server {
```

Comment:
Nit, can we make it more obvious that this is a constructor? Maybe `newHTTPMock`?